opsramp/base.py

```diff
@@ -100,10 +146,8 @@ def __init__(self, url, auth, tracker=None, session=None):
             self.tracker = tracker
         else:
             self.tracker = PathTracker()
-        if session:
-            self.session = session
-        else:
-            self.session = requests.Session()
+
+        self.session = Helpers.session_add_retry_handler(session=session)
```

Comment:
+1 This is exactly what I was thinking. I would probably have done the "if session" at this level but not a big deal and I can see this way matches the original blog post.